### PR TITLE
feat(pending-actions): persona-agnostic aggregator + 5-item display cap

### PR DESCRIPTION
**Pull request by manishdixitlfx**
## Summary

Two related changes to **Pending Actions** on the dashboards:

1. **Persona-agnostic** — every user sees the full set of actions they owe, regardless of which dashboard/persona they're on. Previously gated to board members only; other personas received an empty array.
2. **Display cap + hide-aware refill** — the component now renders up to 5 items at a time. When the user dismisses one (the existing 24h hide cookie), the next unseen action slides in. Also fixes a latent bug where the hide cookie was being set but never read.

### Backend (`user.service.ts`, `project.service.ts`)

- **`getPendingActions`** no longer dispatches on `persona`. The param is kept on the signature for back-compat with all four existing callers (multi-persona, user-dashboard, board-member, executive-director dashboards) but is ignored.
- **`getBoardMemberActions` → `getUserPendingActions`**. Dropped the `meeting_type === 'board'` filter so working-group and non-board meetings within 2 weeks also surface.
- **`getPendingActionSurveys`** (Snowflake): dropped the `AND COMMITTEE_CATEGORY = 'Board'` clause. Non-responded surveys across any committee the user is eligible for now return. The per-row `EMAIL = ?` filter still scopes to the user — no cross-user leakage.
- **New optional `?limit=` query param** on `GET /api/user/pending-actions` (clamped 1..100). Defaults to unbounded for back-compat. Backend work is unchanged — the aggregator still runs in full and then slices. The win is over-the-wire payload size when a caller wants a summary view.

### Frontend (`pending-actions.component.ts` / `.html`)

- New `displayLimit = input<number>(5)` input.
- New `visibleActions` computed: filters items where `HiddenActionsService.isActionHidden(item)` returns true, then slices to `displayLimit`.
- Template loops `visibleActions()` with `track item` (was `pendingActions()` with `track $index`).
- Result: focused view of 5 actions. Click dismiss → cookie set → `visibleActions` recomputes → item drops out → next unseen item fills the slot.
- **Fixed**: the 24h hide cookie was being set by `handleActionClick` but no code ever read it back, so dismissed items were reappearing. Now consistent.

## Why persona-agnostic

A pending action is something the user owes the system. It shouldn't depend on which dashboard they're viewing. The board-only scoping was an MVP artifact, not a principled design choice — and it broke for users with multiple personas (e.g., someone who's both a board member and a maintainer only saw board-side actions on their board dashboard, nothing on maintainer).

## Why frontend cap vs server-side pagination

Backend work is already user-scoped (Snowflake filters by email; meeting fetch is batched via #529). Limiting the response to 5 items doesn't reduce backend work — it only reduces wire bytes. Full server-side pagination (offset/cursor + hide-cookie coordination) is over-engineering at this data volume (typical user <20 pending actions). Frontend cap achieves the UX goal; the optional `?limit=` handles the wire-bytes case if anyone needs it.

## Protected file

`apps/lfx-one/src/server/services/project.service.ts` is on the LFX preflight protected list. The change is a minimal one-line WHERE clause removal (`COMMITTEE_CATEGORY = 'Board'`). Flagging for code-owner attention.

## What's NOT in this PR (deferred)

- New action sources: pending votes, pending RSVPs, mailing-list moderation, PR-review backlogs. Phase 2 — filed separately if wanted.
- Server-side pagination with offset/cursor. Not justified by data volume.
- A `source: 'meeting' | 'survey'` discriminator field on `PendingActionItem`. Frontend doesn't need it today.

## Test plan

- [ ] As a **board member**: dashboard shows board meetings + board surveys + non-board meetings/surveys you're registered on, up to 5 total.
- [ ] As a **maintainer or contributor**: dashboard now shows pending actions (previously empty).
- [ ] As an **ED**: same as board member — all actions in scope.
- [ ] Dismiss one action → cookie is set → item disappears → next unseen item fills the slot (no page refresh).
- [ ] Refresh within 24h → dismissed items stay hidden.
- [ ] After 24h → dismissed items reappear (cookie expired).
- [ ] `GET /api/user/pending-actions?limit=3&persona=X&projectUid=Y&projectSlug=Z` → response has at most 3 items.
- [ ] Existing callers (no `limit` param) continue to receive the full list.
- [ ] Server logs show `get_pending_actions` success with correct `action_count` and optional `limit` metadata.

## Commits

- `ed997b22` feat(pending-actions): make persona-agnostic and drop board-only filters
- `0a543e23` feat(pending-actions): cap display to 5 items with hide-aware refill

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/lfx-one/src/app/modules/dashboards/components/pending-actions/pending-actions.component.ts [hunk 1]
```diff
@@ -17,9 +17,20 @@ export class PendingActionsComponent {
...
+  // the next change-detection pass. The full list still lives on the parent — this just controls
+  // what gets rendered.
+  protected readonly visibleActions: Signal<PendingActionItem[]> = computed(() => {
+    const unhidden = this.pendingActions().filter((item) => !this.hiddenActionsService.isActionHidden(item));
+    return unhidden.slice(0, this.displayLimit());
+  });
+
   protected handleActionClick(item: PendingActionItem): void {
     this.hiddenActionsService.hideAction(item);
     this.actionClick.emit(item);
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/app/modules/dashboards/components/pending-actions/pending-actions.component.ts:48]: Done — added `hiddenActionsVersion = signal(0)` and bump it in `handleActionClick` so the computed recomputes immediately on dismiss. Reading the version signal inside `visibleActions` registers the dependency.

See commit 0bf30778.

File: apps/lfx-one/src/server/services/user.service.ts [hunk 2]
```diff
@@ -458,12 +458,21 @@ export class UserService {
...
    */
-  public async getPendingActions(req: Request, persona: PersonaType, projectUid: string, email: string, projectSlug: string): Promise<PendingActionItem[]> {
-    if (persona === 'board-member') {
-      return this.getBoardMemberActions(req, email, projectSlug, projectUid);
-    }
-    // Future personas: maintainer, contributor can be added here
-    return [];
+  public async getPendingActions(
+    req: Request,
+    persona: PersonaType,
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/server/services/user.service.ts]: Done — rewrote the docblock to describe the persona-agnostic aggregator, dropped the stale `@param persona` line, and documented what `limit` actually does.

See commit 0bf30778.

File: apps/lfx-one/src/server/services/user.service.ts [hunk 3]
```diff
@@ -458,12 +458,21 @@ export class UserService {
...
+    projectUid: string,
+    email: string,
+    projectSlug: string,
+    limit?: number
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/server/services/user.service.ts]: Done — removed the `persona` parameter (and the `PersonaType` import) entirely since it's truly unused. The controller still validates `?persona=` as a required query param, so the public API contract is unchanged; it just no longer forwards persona into the service.

See commit 0bf30778.

File: apps/lfx-one/src/app/modules/dashboards/components/pending-actions/pending-actions.component.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Copyright The Linux Foundation and each contributor to LFX.
 // SPDX-License-Identifier: MIT
 
-import { Component, inject, input, output } from '@angular/core';
+import { Component, computed, inject, input, output, Signal } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/app/modules/dashboards/components/pending-actions/pending-actions.component.ts]: Done — added a `hiddenActionsVersion` signal bumped in `handleActionClick` and read inside the `visibleActions` computed, using the exact pattern you suggested. Dismissals now invalidate the window immediately and the next waiting item slides in on the same change-detection pass.

See commit 0bf30778.

File: apps/lfx-one/src/app/modules/dashboards/components/pending-actions/pending-actions.component.ts [hunk 5]
```diff
@@ -17,9 +17,20 @@ export class PendingActionsComponent {
...
   public readonly actionClick = output<PendingActionItem>();
 
+  // Windowed view of the incoming list: drop anything the user has dismissed (HiddenActionsService
+  // sets a 24h cookie on click) and cap to displayLimit so the user focuses on a handful at a time.
+  // When one is dismissed, it leaves the window and the next waiting item slides into the slot on
+  // the next change-detection pass. The full list still lives on the parent — this just controls
+  // what gets rendered.
+  protected readonly visibleActions: Signal<PendingActionItem[]> = computed(() => {
+    const unhidden = this.pendingActions().filter((item) => !this.hiddenActionsService.isActionHidden(item));
+    return unhidden.slice(0, this.displayLimit());
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/app/modules/dashboards/components/pending-actions/pending-actions.component.ts]: Good catch — the upstream slicing was in all four callers, not just user-dashboard. Dropped the `.filter(hidden).slice(0, N)` from user-dashboard, board-member-dashboard, executive-director-dashboard, and multi-persona-dashboard. Each now passes the raw aggregated list and uses `[displayLimit]="2"` on the three compact dashboard cards, default 5 on multi-persona. Windowing (dismiss filtering + refill + cap) lives in one place now — `PendingActionsComponent`.

See commit 0bf30778.

**Comment by manishdixitlfx** [comment]
## Review Feedback Addressed

Commit: 0bf30778

### Changes Made
- **pending-actions.component.ts**: added `hiddenActionsVersion = signal(0)` bumped in `handleActionClick` and read in `visibleActions` computed — dismissals now recompute the window immediately instead of waiting for the parent to re-supply the array (per @coderabbitai + @copilot-pull-request-reviewer)
- **user-dashboard / board-member-dashboard / executive-director-dashboard / multi-persona-dashboard**: dropped the upstream `.filter(hidden).slice(0, N)` that was bypassing the child component's `displayLimit` and refill behavior; each caller now passes the raw aggregated list with `[displayLimit]="2"` on the three compact dashboard cards (default 5 on multi-persona). Windowing lives in one place now — `PendingActionsComponent` (per @coderabbitai)
- **user.service.ts**: rewrote `getPendingActions` docblock to describe the persona-agnostic aggregator and documented `limit`; dropped the unused `persona` parameter and `PersonaType` import (per @copilot-pull-request-reviewer)
- **user.controller.ts**: stopped forwarding persona into the service call; `?persona=` query-param validation stays in the controller so the public API contract is unchanged for callers (per @copilot-pull-request-reviewer)

### Threads Resolved
5 of 5 unresolved threads addressed in this iteration.

**Comment by dealako** [comment]
@manishdixitlfx — review pass on tip `0bf30778`.

**Prior AI feedback recap:** All 5 threads from CodeRabbit ([#1](https://github.com/linuxfoundation/lfx-v2-ui/pull/539#discussion_r3126315819), [#2](https://github.com/linuxfoundation/lfx-v2-ui/pull/539#discussion_r3126315827)) and Copilot ([#1](https://github.com/linuxfoundation/lfx-v2-ui/pull/539#discussion_r3126303457), [#2](https://github.com/linuxfoundation/lfx-v2-ui/pull/539#discussion_r3126303490), [#3](https://github.com/linuxfoundation/lfx-v2-ui/pull/539#discussion_r3126303508)) were resolved in `0bf30778` — no overlapping comments below.

---

## 🔴 High — user-dashboard still gatekeeps by persona (contradicts the test plan)

**`user-dashboard.component.ts:31,51,75–98`**

`rawContributorActions = signal<PendingActionItem[]>([])` is declared but never written to. `pendingActions` (line 51) selects between `rawBoardActions()` and `rawContributorActions()` based on `isBoardScoped()`. `initBoardActions()` short-circuits at line 85:

```ts
if (!isBoardScoped || !project?.slug || !project?.uid) {
  return of([]); // ← non-board personas always hit this
}
```

Net effect: maintainers and contributors on the **user dashboard** still see an empty pending-actions list despite the backend now returning their actions. The backend change is wired correctly, but the frontend wiring wasn't completed.

The PR test plan says:
> As a **maintainer or contributor**: dashboard now shows pending actions (previously empty).

This is not true in the current code. Suggested resolution (pick one):

1. **Finish the wiring** — remove the `isBoardScoped` gate in `initBoardActions`, call `getPendingActions` for all personas, and delete the stale `rawContributorActions` placeholder.
2. **Scope the PR description** — note that user-dashboard remains board-only in this iteration and file a follow-up ticket for non-board personas.

---

## 🟡 Medium — `getPendingActionSurveys` does not normalize email

**`project.service.ts:974–988`**

```ts
// Current — email passes through as-is
WHERE EMAIL = ?    -- binds: email
```

Every other method in this file that queries by email normalizes first (`email.trim().toLowerCase()` — lines 443, 534, 677). If Auth0 yields a mixed-case email, the Snowflake query silently returns zero surveys for that user.

```ts
// Suggested fix
const normalizedEmail = email.trim().toLowerCase();
// pass normalizedEmail to the query binds
```

---

## 🟡 Medium — `@for` tracking by object identity re-mounts every row on refresh

**`pending-actions.component.html:18`**

```html
@for (item of visibleActions(); track item)
```

`PendingActionItem` has no stable `id` field (`components.interface.ts:474`). After a `refresh$` emission the service returns new object references, so Angular tears down and recreates every row — the same effect as `track $index`, just at a different phase.

`HiddenActionsService` already computes a stable identifier (DJB2 of `buttonLink ?? type-badge-text`). A simpler option is a composite track expression directly in the template:

```html
@for (item of visibleActions(); track (item.buttonLink ?? item.type + '-' + item.text))
```

Or add an `id` field to `PendingActionItem` populated by the aggregator.

---

## 🟢 Low — `?persona=` is validated but never consumed; log metadata may mislead

**`user.controller.ts:42–52, 95`**

`persona` is still a hard-required query param (400 if absent) and is always included in the `logger.success` metadata, but is never forwarded to the service. The inline comment acknowledges the API-contract intent, which is reasonable. Worth a quick decision: is this a permanent vestigial param, or is there a planned deprecation path? If permanently vestigial, dropping it from the contract in a follow-up keeps the surface honest. Either way, excluding it from the success-log metadata (or adding a comment that it's logged for client-debugging only) avoids confusion when someone reads the logs and wonders why `persona` appears but doesn't affect the response.

---

## 🟢 Low — recurring-meeting deduplication edge case

**`user.service.ts` → `transformMeetingsToActions`**

A recurring meeting with two active occurrences within the 2-week window produces two identical-looking rows (same `type`, `text`, `buttonLink`). Since `HiddenActionsService.getActionIdentifier` keys on `buttonLink`, dismissing one hides both — which could be the desired behavior ("I don't want to see this meeting series") but might also surprise a user who expected to dismiss only the nearer occurrence. Worth an explicit product decision; if each occurrence should be independently dismissible, the identifier needs an occurrence date component.

---

## 🟢 Low — `displayLimit` accepts `0` and negative values

**`pending-actions.component.ts:20`**

```ts
public readonly displayLimit = input<number>(5);
```

A caller passing `[displayLimit]="0"` silently renders nothing. A one-line guard in `visibleActions` makes the misuse explicit:

```ts
return unhidden.slice(0, Math.max(1, this.displayLimit()));
```

---

## 🟢 Low — `getUserPendingActions` docblock over-claims scope

**`user.service.ts:865–870`**

"Aggregate pending actions for the current user within a project scope" is accurate, but the opening line "across every persona" could be read as "user-global." The aggregation is persona-agnostic *within a project* (still scoped to `projectSlug`/`projectUid`). Tightening the wording avoids confusion for future callers.

---

## 🟢 Low — no test coverage for the new logic

No spec files exist for `PendingActionsComponent`, `UserService.getPendingActions`, `UserController.getPendingActions`, or `HiddenActionsService`. The new `displayLimit` input, `hiddenActionsVersion` signal counter, `limit` query-param clamping, and the `limit > 0` branch are all untested.

Given the prior CodeRabbit/Copilot round specifically pushed for the dismiss-refill behavior, locking it in with a unit test would prevent a regression. At minimum: a test for the controller's `limit` clamping (valid, zero, > 100, non-numeric) and one for the component's `visibleActions` recompute after `handleActionClick`.

---

🤖 Generated with [Claude Code](https://claude.ai/code)

File: apps/lfx-one/src/server/controllers/user.controller.ts [hunk 6]
```diff
@@ -77,14 +77,22 @@ export class UserController {
...
       }
 
-      // Get pending actions from service
-      const pendingActions = await this.userService.getPendingActions(req, persona, projectUid, userEmail, projectSlug);
+      // Optional `?limit=` clamps the response size so mobile / summary cards can request a
+      // smaller payload. Backend work isn't reduced (the full list is computed and then sliced);
+      // the win is over-the-wire bytes. Unbounded by default for back-compat with existing clients.
+      const limitParam = parseInt(req.query['limit'] as string, 10);
+      const limit = Number.isFinite(limitParam) && limitParam > 0 ? Math.min(limitParam, 100) : undefined;
+
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/server/controllers/user.controller.ts]: Done — made `?limit=` validation explicit. When the param is present we now reject arrays, non-numeric strings, and non-positive integers with a `ServiceValidationError` instead of silently falling back to unbounded. Missing param still defaults to unbounded for back-compat. The `String(parsed) !== limitQuery.trim()` check also guards against values like `10abc` that `parseInt` would otherwise accept as `10`.

See commit 6106da97.

File: apps/lfx-one/src/app/modules/dashboards/components/pending-actions/pending-actions.component.ts [hunk 7]
```diff
@@ -17,11 +17,30 @@ export class PendingActionsComponent {
...
+
+  // Windowed view of the incoming list: drop anything the user has dismissed (HiddenActionsService
+  // sets a 24h cookie on click) and cap to displayLimit so the user focuses on a handful at a time.
+  // When one is dismissed, it leaves the window and the next waiting item slides into the slot on
+  // the next change-detection pass. The full list still lives on the parent — this just controls
+  // what gets rendered.
+  protected readonly visibleActions: Signal<PendingActionItem[]> = computed(() => {
+    this.hiddenActionsVersion();
+    const unhidden = this.pendingActions().filter((item) => !this.hiddenActionsService.isActionHidden(item));
+    return unhidden.slice(0, this.displayLimit());
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/app/modules/dashboards/components/pending-actions/pending-actions.component.ts]: Done — clamped `displayLimit()` to `>= 0` (falling back to `5` for non-finite values) before passing it to `slice`. A parent component that accidentally binds a negative number will no longer cause `slice(0, -N)` to silently drop the last N items; it will render an empty list instead.

See commit 6106da97.

**Comment by manishdixitlfx** [comment]
## Review Feedback Addressed

Commit: 6106da97

### Changes Made
- **user.controller.ts**: validate `?limit=` explicitly — arrays, non-numeric strings, and non-positive integers now return a `ServiceValidationError` instead of silently defeating the cap. Missing param still means unbounded for back-compat (per @copilot-pull-request-reviewer)
- **pending-actions.component.ts**: clamp `displayLimit()` to `>= 0` (fallback 5 for non-finite) before slicing so a negative input can't cause `slice(0, -N)` to drop items from the end (per @copilot-pull-request-reviewer)

### Threads Resolved
2 of 2 unresolved threads addressed in this iteration.

File: apps/lfx-one/src/server/controllers/user.controller.ts [hunk 8]
```diff
@@ -77,14 +77,38 @@ export class UserController {
...
+            ServiceValidationError.forField('limit', 'limit query parameter must be a positive integer', {
+              operation: 'get_pending_actions',
+              service: 'user_controller',
+              path: req.path,
+            })
+          );
+          return;
+        }
+        limit = Math.min(parsed, 100);
+      }
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/server/controllers/user.controller.ts:100]: Done — switched the gate from `typeof === 'string'` to `!== undefined` and coerce non-strings to `NaN` via the ternary, so arrays (`?limit=1&limit=2`) and objects (`?limit[foo]=1`) now fail the `Number.isInteger` check and hit the same `ServiceValidationError` branch. Absent param still means unbounded. Net 1 extra line vs the over-slim version.

See commit 3c290701.

File: apps/lfx-one/src/server/controllers/user.controller.ts [hunk 9]
```diff
@@ -77,14 +77,38 @@ export class UserController {
...
+      // smaller payload. Absent = unbounded; when present it must be a positive integer so
+      // malformed values can't silently defeat the cap. `Number.isInteger` rejects NaN,
+      // floats, and `"10abc"` — no separate array branch because Express typing already
+      // tells us it's `string | string[] | ParsedQs | …`, and non-strings fall out.
+      const limitQuery = req.query['limit'];
+      let limit: number | undefined;
+      if (typeof limitQuery === 'string') {
+        const parsed = Number(limitQuery);
+        if (!Number.isInteger(parsed) || parsed <= 0) {
+          next(
```
[the remark this answers is not in the document]
Comment (manishdixitlfx) [reply apps/lfx-one/src/server/controllers/user.controller.ts]: Done — switched the gate from `typeof === 'string'` to `!== undefined` and coerce non-strings to `NaN` via the ternary, so arrays (`?limit=1&limit=2`) and objects (`?limit[foo]=1`) now fail the `Number.isInteger` check and hit the same `ServiceValidationError` branch. Absent param still means unbounded. Net 1 extra line vs the over-slim version.

See commit 3c290701.

**Comment by manishdixitlfx** [comment]
## Review Feedback Addressed

Commit: 3c290701

### Changes Made
- **user.controller.ts**: the previous slim-down over-rotated — using `typeof === 'string'` as the gate meant arrays and objects fell through to unbounded. Switched to `!== undefined` with a `typeof === 'string' ? Number(...) : NaN` coercion so non-strings now fail the `Number.isInteger` check and hit the same `ServiceValidationError` branch. Absent param still means unbounded (per @coderabbitai, @copilot-pull-request-reviewer)

### Threads Resolved
2 of 2 unresolved threads addressed in this iteration.

**Comment by dealako** [comment]
@manishdixitlfx — follow-up pass on tip `3c290701`.

The three new commits (`6106da97`, `adaf91a9`, `3c290701`) address the `?limit=` controller validation and the `displayLimit` guard in the component. The iterative refinement on the `?limit=` parsing is good — `!== undefined` + `non-string → NaN` coercion correctly covers arrays, objects, floats, and empty strings with a single `ServiceValidationError` branch.

Here's the full scorecard against my earlier review:

| # | Severity | Finding | Status |
|---|---|---|---|
| 1 | 🔴 High | `user-dashboard.component.ts` still gatekeeps by persona — `rawContributorActions` never written to | 🔴 **Open** |
| 2 | 🟡 Med | `getPendingActionSurveys` doesn't normalize email before Snowflake `WHERE EMAIL = ?` | 🟡 **Open** |
| 3 | 🟡 Med | `@for` tracking by object identity — full DOM remount on every refresh | 🟡 **Open** |
| 4 | 🟢 Low | `?persona=` still required (400) and always logged but never consumed downstream | 🟢 **Open** (acknowledged as intentional) |
| 5 | 🟢 Low | Recurring-meeting dedup edge case | 🟢 **Open** (product decision needed) |
| 6 | 🟢 Low | `displayLimit` accepts invalid values | ✅ **Resolved** in `6106da97` |
| 7 | 🟢 Low | No test coverage for new logic | 🟢 **Open** |

---

### Items worth resolving before merge

**#1 — user-dashboard still gatekeeps** ← same high-priority flag from before

`rawContributorActions = signal<PendingActionItem[]>([])` at `user-dashboard.component.ts:31` is still never written to. The `pendingActions` signal on line 51 returns it for non-board personas, so maintainers and contributors permanently see an empty list. The backend aggregator is persona-agnostic now, but the frontend wiring stops at the board path.

The test plan says maintainers/contributors "now shows pending actions." This still won't pass.

**#2 — email normalization in `getPendingActionSurveys`**

`project.service.ts:985` (`WHERE EMAIL = ?`) still passes the raw `email` argument. All sibling Snowflake methods in that file normalize via `.trim().toLowerCase()` (lines 443, 534, 677). One-liner fix:

```ts
const normalizedEmail = email.trim().toLowerCase();
```

**#3 — `@for` tracking by object identity**

`track item` on `pending-actions.component.html:19` still tracks by reference. After any `refresh$` emission, every row is torn down and recreated. A stable composite works today without an interface change:

```html
@for (item of visibleActions(); track (item.buttonLink ?? item.type + '-' + item.text))
```

---

### Low-priority items (can defer to follow-up)

- **#4** — `?persona=` is required but unused. Marking it deprecated or optional (with a `LFXV2-` follow-up ticket) would keep the contract honest.
- **#5** — Recurring meetings: two occurrences of the same meeting within 2 weeks produce duplicate rows; dismissing one hides both. Product decision needed.
- **#7** — No spec coverage for the new `displayLimit` input, `hiddenActionsVersion` counter, and `limit` clamping. Good candidates for unit tests before this ships.

---

🤖 Generated with [Claude Code](https://claude.ai/code)

**Comment by manishdixitlfx** [comment]
@dealako — thanks for the thorough second pass. Pushed a follow-up covering 3 items (1 cherry-pick + 2 direct fixes) and leaving 3 as explicit responses.

Tip now at **6a773de8**.

### Resolved in this iteration

**#1 — user-dashboard gatekeeping** (high)
Cherry-picked `c9c42887` from the Phase 2 follow-up branch so #539 ships self-consistent. `user-dashboard.component.ts` no longer gates on `isBoardScoped()` — the `rawContributorActions` always-empty signal is gone, and the dashboard calls `getPendingActions` for any persona once a project context is available. Maintainers and contributors will now populate the card.

**#2 — email normalization in `getPendingActionSurveys`**
Added `const normalizedEmail = email.trim().toLowerCase()` before the Snowflake query and pass that to the binding. Matches the sibling pattern in `resolveEmailToSub`, `resolveEmailToUsername`, etc. Commit 6a773de8.

**#3 — `@for` tracking**
Swapped `track item` for `track item.buttonLink ?? item.type + '-' + item.text` so rows match across `refresh$` emissions instead of remounting. Stable composite avoids adding a new field to the shared interface. Commit 6a773de8.

### Deferred with rationale

**#4 — `?persona=` required but unused**
Keeping the validation in place for API-contract stability with existing callers (multi-persona + the three single-persona dashboards still send it). Agree that a cleaner deprecation path is the right long-term move — I'll file an `LFXV2-` ticket to drop the requirement + update all call sites in one follow-up so the removal lands atomically.

**#5 — recurring meeting dedup**
Genuine product decision, not a code hygiene issue. Current behavior: each active occurrence within the 2-week window produces its own row, and dismissing a row hides that specific occurrence via the `HiddenActionsService` cookie. Two reasonable treatments — collapse to one "next occurrence" action per meeting, or keep per-occurrence and let hide cascade. Deferring until we have a product call.

**#7 — no spec coverage**
Fair — the new `displayLimit` input, `hiddenActionsVersion` counter, and `?limit=` clamping are all testable surface. Will add in a follow-up PR so this one doesn't grow further; none are covered today in the component's existing test file either, which is a separate gap.

🤖 Generated with [Claude Code](https://claude.ai/code)

**Comment by dealako** [comment]
@manishdixitlfx — final pass on tip `6a773de8`. All material issues are closed.

## Scorecard

| # | Severity | Finding | Status |
|---|---|---|---|
| 1 | 🔴 High | user-dashboard still gatekeeps by persona | ✅ **Resolved** in `c9c42887` |
| 2 | 🟡 Med | `getPendingActionSurveys` missing email normalization | ✅ **Resolved** in `6a773de8` |
| 3 | 🟡 Med | `@for` tracking by object identity | ✅ **Resolved** in `6a773de8` |
| 4 | 🟢 Low | `?persona=` required but unused | ⏭️ **Deferred** — LFXV2 follow-up ticket planned |
| 5 | 🟢 Low | Recurring-meeting dedup edge case | ⏭️ **Deferred** — product decision pending |
| 6 | 🟢 Low | `displayLimit` accepts invalid values | ✅ **Resolved** in `6106da97` |
| 7 | 🟢 Low | No spec coverage for new logic | ⏭️ **Deferred** — follow-up PR planned |

## Notes on the fixes

**#1 (`c9c42887`)** — Clean removal. `rawContributorActions` is gone, `combineLatest([project$, isBoardScoped$])` collapsed to a simple `project$.pipe(switchMap(...))`. `isBoardScoped` correctly stays for the UI label computations only. Unused `signal` and `combineLatest` imports cleaned up as well.

**#2 (`6a773de8`)** — `const normalizedEmail = email.trim().toLowerCase()` before the Snowflake bind, consistent with the pattern at lines 443, 534, 677 in the same file. Good call documenting the column case assumption in the comment.

**#3 (`6a773de8`)** — `track item.buttonLink ?? item.type + '-' + item.text` matches the stable composite that `HiddenActionsService.getActionIdentifier` already uses. No interface change needed.

## Overall

The PR is in good shape. The backend is persona-agnostic, the display cap + refill behavior is correctly owned by `PendingActionsComponent`, and all three callers that needed fixing are updated. The three deferred items all have clear rationale and follow-up paths — none are blockers for shipping.

✅ **LGTM**

🤖 Generated with [Claude Code](https://claude.ai/code)

**Review by dealako — APPROVED** [review]
Final review passes. See comments above. Approved.
